contracts/lib/SolidMath.sol

```diff
@@ -0,0 +1,56 @@
+// SPDX-License-Identifier: BUSL-1.1
+
+pragma solidity 0.8.13;
+
+/**
+ * @dev Solid DAO Math Operations and Constants.
+ */
+abstract contract SolidMath {
+
+    uint256 constant public WEEKS_IN_SECONDS = 604800;
+
+    /**
+    @notice BASIS define in which basis value the interest rate per week must be informed
+    */
+    uint256 constant BASIS = 1000000;
+
+    /**
+     * @dev Calculates the number of weeks between a period
+     * @param _initialDate uint256 initial date to be informed in seconds
+     * @param _contractExpectedDueDate uint256 the carbon credit contract expected due date to be informed in seconds
+     */
+    function weeksInThePeriod(uint256 _initialDate, uint256 _contractExpectedDueDate) public pure returns(bool, uint256) {
+        if (_contractExpectedDueDate < WEEKS_IN_SECONDS) {
+            return (false, 0);
+        }
+        if (_initialDate < WEEKS_IN_SECONDS) {
```

Comment:
I don't get this condition as well.

Reply:
Let's say a dev make a mistake and for some reason sent the contract due date in a period less than a week, it should return false. Pay attention the basis is seconds so might have miscalculation for who will call this function. Then, it's weeksInThePeriod function responsibility to check all parameters it receives.